From ff4fb46407f3aee09432a897822e4934d5c6b3ff Mon Sep 17 00:00:00 2001
From: Jingchang Lu <b35083@freescale.com>
Date: Thu, 4 Aug 2011 09:59:54 +0800
Subject: [PATCH 44/52] Fix Max3353 otg toggle bug

The max3353 charge pump work with SOC FS/LS transceiver,
during dual role toggle, the root hub port should be powered
off to disable port status detection when suspend host mode,
otherwise it won't work properly when switch to host mode.

Signed-off-by: Jingchang Lu <b35083@freescale.com>
---
 drivers/usb/host/ehci-fsl.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

--- a/drivers/usb/host/ehci-fsl.c
+++ b/drivers/usb/host/ehci-fsl.c
@@ -528,7 +528,7 @@ static int ehci_fsl_suspend(struct devic
 	pdata->pm_portsc &= cpu_to_hc32(ehci, ~PORT_RWC_BITS);
 
 	pdata->suspended = 1;
-#if 0
+#if defined(CONFIG_USB_M5441X_MAX3353_FSLS)
 	/* clear PP to cut power to the port */
 	tmp = ehci_readl(ehci, &ehci->regs->port_status[0]);
 	tmp &= ~PORT_POWER;
